buttercup.el

```diff
@@ -41,6 +41,56 @@
 
 (require 'cl-lib)
 (require 'buttercup-compat)
+(require 'format-spec)
+(require 'ert nil t)
+(require 'warnings)
+
+;;;;;;;;;;;;;;;;;;;;;;;;
+;;; closure manipulation
+
+(defun buttercup--enclosed-expr (x)
+  "Given a zero-arg closure, return its unevaluated expression.
```

Comment:
Calling the argument CLOSURE would likely be more clear, no? (I'm still impressed this works even for byte-compiled stuff …)

Reply:
Well, the symbol `closure` already has a special meaning: it is the replacement for `lambda` in a lexical closure, although unlike `lambda`, it does not create a self-quoting expression. So it would be kind of like taking a function argument and naming it `lambda`. I think it would technically work, but it would be kind of confusing. Although now that this function works on both closures and lambdas, I guess I could just call the argument FUN. I think I'll do that.

As for working on byte-compiled things, as far as I know that shouldn't work. Figuring out how lexical scoping and macro expansion work together is tricky, but my understanding is: the `expect` macro is expanded inside a test spec. The expanded expression includes a series of lambda expressions, and is uncompiled. This happens regardless of whether buttercup.el itself is compiled. Then those lambdas are converted to closures at some point by the emacs lisp interpreter. It's possible that if you byte-compiled the test file itself, these closures would get macroexpanded and then byte-compiled and would break as a result. I'll try this and see what happens.

Reply:
I've switched to more accurate naming for functions and arguments in 2f97656. The argument to this and similar functions is now FUN, and all references to closures in function names and docstrings are replaced with "function" or "wrapper function".